server/controllers/Builders.js

```diff
@@ -0,0 +1,111 @@
+const Builder = require('../db/models/Builder');
+const mongoose = require('mongoose');
+
+//CREATE BUILDER PROFILE//
+
+exports.addProfile = async (req, res) => {
+  try {
+    const builder = await new Builder({
+      ...req.body,
+      owner: req.user._id,
+    });
+    await builder.save();
+    res.status(200).send(builder);
+  } catch (error) {
+    res.status(400).json({ error: error.message });
+  }
+};
+
+//GET 'ALL' BUILDER PROFILES//
+
+exports.getAllBuilderInfo = async (req, res) => {
+  try {
+    const builder = await Builder.find({ owner: req.user._id });
+    if (!builder)
+      return res.status(400).json({ message: 'User Info not found' });
+    res.status(200).json(builder);
+  } catch (error) {
+    res.status(400).json({ error: error.message });
+  }
+};
+
+//GET ONE BUILDER PROFILE//
+
+exports.getSpecificBuilderInfo = async (req, res) => {
+  const _id = req.params.id;
+
+  if (!mongoose.Types.ObjectId.isValid(_id))
+    return res.status(400).json({ message: 'not valid info' });
+
+  try {
+    const builder = await Builder.findOne({ _id, owner: req.user._id });
+    if (!builder)
+      return res.status(400).json({ message: 'User Info not found' });
+    res.status(200).json(builder);
+  } catch (error) {
+    res.status(400).json({ error: error.message });
+  }
+};
+
+//UPDATE BUILDER PROFILE//
+
+exports.updateBuilder = async (req, res) => {
+  const updates = Object.keys(req.body);
+  const allowedUpdates = [
+    'school',
+    'technologies',
+    'role',
+    'specialty',
+    'level',
+    'level',
```

Comment:
What are the two levels?

Reply:
duplicated value by mistake. It's fixed.